llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7060,6 +7060,49 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
       return new ICmpInst(I.getSwappedPredicate(Pred), Builder.CreateXor(X, Y),
                           Z);
 
+    // Transform X s<  ~X  -->   X s<  0
+    //           X s>  ~X  -->   X s>  -1
+    //           X s>= ~X  -->   X s>  -1
+    //           X s<= ~X  -->   X s<  0
+    //           X u<  ~X  -->   X s>  -1
+    //           X u>  ~X  -->   X s<  0
+    //           X u<= ~X  -->   X s>  -1
+    //           X u>= ~X  -->   X s<  0
+    //           X ==  ~X  -->   false
+    //           X !=  ~X  -->   true
+    if (match(&I, m_c_ICmp(Pred, m_Value(X), m_Not(m_Deferred(X))))) {
+      CmpInst::Predicate PredEqRemoved;
```

Comment:
```suggestion
      CmpInst::Predicate NewPred;
```
For the unsigned case, this isn't really PredEqRemoved anymore.

Reply:
right. I missed it.